src/Components/WebAssembly/testassets/ThreadingApp/Pages/Counter.razor

```diff
@@ -0,0 +1,79 @@
+﻿@page "/counter"
+@using System.Runtime.InteropServices
+
+<h1>Counter</h1>
+
+<p>Current count: @currentCount</p>
+
+<button class="btn btn-primary" @onclick="IncrementCount">Click me</button>
+
+@code {
+    int currentCount = 0;
+
+    void IncrementCount()
+    {
+        currentCount++;
+    }
+
+    protected override async Task OnInitializedAsync()
+    {
+        if(!OperatingSystem.IsBrowser())
+        {
+            return;
+        }
+
+        if (Thread.CurrentThread.ManagedThreadId != 1)
+        {
+            throw new Exception("We should be on main thread!");
+        }
+
+        Exception exc = null;
+        try
+        {
+            // send me to the thread pool
+            await Task.Delay(10).ConfigureAwait(false);
+            StateHasChanged(); // render should throw
+        }
+        catch(Exception ex)
+        {
+            exc=ex;
+            Console.WriteLine(ex.Message);
+        }
+        if (exc == null || exc.Message != "The current thread is not associated with the Dispatcher. Use InvokeAsync() to switch execution to the Dispatcher when triggering rendering or component state.")
```

Comment:
this message will probably get localized, right? could we assert on something else about it, like maybe that it's a specific type of exception?

Reply:
All the E2E unit tests run in US locale. There are also other places with the same assert. I unified the style to match others closely.